### PR TITLE
Update simulation option on Wiki

### LINKED ISSUE #372: Out of date simulation option wiki
@ChunYen-Chen The revised wiki looks MUCH better. Excellent jobs!

### Major comment
- [ ] The current wiki does not list the options available in `configure.py`, instead relying on `python3 configure.py -lh`. It can be inconvenient since it requires downloading GAMER to view the options. Moreover, we still refer to `Installation:-Simulation-Options.md` throughout the wiki. So I suggest updating `Installation:-Simulation-Options.md` to address this problem. For instance, in each table, we could add a column next to `Option` to display the corresponding configuration option (e.g., `--model` for `MODEL`). But let's do that in a separate PR.

### Minor comments
- [x] On the top of `Makefile_base`, replace `compiler and flags` with `compilers and flags` to match line 35.
- [x] How about adding `Initial-Conditions.md` to the sidebar (e.g., below `GPU`)? Currently it's only listed on the page `Runtime Parameters`, even though it contains  a wealth of additional information.

_Originally posted by @hyschive in https://github.com/gamer-project/gamer/pull/332#pullrequestreview-2425681959_

### PR DESCRIPTION
**Pull request by ChunYen-Chen**
* Remove all the editing instructions of the `Makefile`
* Update the simulation options with `configure.py` options
* Update libyt arguments
* Separate the developer guide of `configure.py` to a new page
* Move simulation options under `Installation: Option List` page
* Add a new page of ELBDM
* Add ELBDM arguments

Resolve #372 and part of #384.

### CODE REVIEW

File: doc/wiki/Installation:-Generate-Makefile.md [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+| `-h`        | -               | Show a short help message. |
+| `-lh`       | -               | Show a detailed help message. |
+| `--machine` | Filename string | Select the `*.config` file under `configs` directory. |
+
+## Physical Modules
+
+<table>
+   <tr>
+      <td align=center rowspan=4 width=130px><b>Option</b></td>
+      <td><b>GAMER Name</b></td>
```
Comment (hyschive) [inline doc/wiki/Installation:-Generate-Makefile.md:54]: This is only relevant for developers. So how about moving it to the last row and renaming it as `Corresponding symbolic constant`?

**Comment by ChunYen-Chen** [comment]
@hyschive I have included the new options introduced by #376.

File: doc/wiki/Installation:-Option-List.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+| `--dual`       | `OFF`, `DE_ENPY`, `DE_EINT`                                | `OFF`   | Enable dual energy formalism | Not supported for `--flu_scheme=RTVD`. `DE_EINT` is not supported yet. | <a name="--dual"></a> `DUAL_ENERGY` |
+| `--mhd`        | `true`, `false`                                            | `false` | Magnetohydrodynamics | - | <a name="--mhd"></a> `MHD` |
+| `--srhd`       | `true`, `false`                                            | `false` | Special relativistic hydrodynamics | Must adopt `--eos=TAUBMATHEWS` | <a name="--srhd"></a> `SRHD` |
+| `--cosmic_ray` | `true`, `false`                                            | `false` | Cosmic rays | Must adopt `--eos=COSMIC_RAY` | <a name="--cosmic_ray"></a> `COSMIC_RAY` |
+| `--eos`        | `GAMMA`, `ISOTHERMAL`, `COSMIC_RAY`, `TAUBMATHEWS`, `USER` | Depend  | [[Equation of state \| equation-of-state]] | The following options only support `GAMMA`: `--flu_scheme=RTVD/CTU`, `--flux=EXACT/ROE`, `--comoving`, `--dual`; see also [[--barotropic \| Installation:-Option-List#--barotropic]] | <a name="--eos"></a> `EOS` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:53]: Currently, we only include links for a small subset of options (e.g., `--barotropic` here). To keep things simple, how about removing all `--XXX` links from this page, as they all refer to content already on the same page?

File: doc/wiki/Installation:-Option-List.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+|:---:|:---:|:---:|---|---|---|
+| `--elbdm_scheme`     | `ELBDM_WAVE`, `ELBDM_HYBRID`                    | `ELBDM_WAVE`    | ELBDM schemes. `ELBDM_WAVE`: wave-only; `ELBDM_HYBRID`: fluid-wave-hybrid-scheme | Must set `--model=ELBDM` | <a name="--elbdm_scheme"></a> `ELBDM_SCHEME` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:61]: "fluid-wave hybrid scheme"

File: doc/wiki/Installation:-Option-List.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+| `--wave_scheme`      | `WAVE_FD`, `WAVE_GRAMFE`                        | `WAVE_FD`       | Wave schemes. `WAVE_FD`: finite difference; `WAVE_GRAMFE`: local spectral method | Must set `--model=ELBDM` | <a name="--wave_scheme"></a> `WAVE_SCHEME` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:62]: "`WAVE_GRAMFE`: local pseudospectral method based on Fourier continuations with Gram polynomials (hereafter also referred to as the Gram-Fourier extension method; GramFE)"

File: doc/wiki/Installation:-Option-List.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+| `--conserve_mass`    | `true`, `false`                                 | `true`          | Enforce the mass conservation | Must set `--model=ELBDM` | <a name="--conserve_mass"></a> `CONSERVE_MASS` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:63]: "Enforce mass conservation"

File: doc/wiki/Installation:-Option-List.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+| `--laplacian_four`   | `true`, `false`                                 | Depend          | Enable the fourth-order Laplacian | Must set `--model=ELBDM` and `--wave_scheme=WAVE_FD` | <a name="--laplacian_four"></a> `LAPLACIAN_4TH` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:64]: "Adopt the fourth-order Laplacian for `--wave_scheme=WAVE_FD`"

File: doc/wiki/Installation:-Option-List.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+| `--gramfe_scheme`    | `GRAMFE_MATMUL`, `GRAMFE_FFT`                   | `GRAMFE_MATMUL` | GramFE schemes. `GRAMFE_MATMUL`: faster for `PATCH_SIZE`=8; `GRAMFE_FFT`: faster for larger patch sizes | Must set `--model=ELBDM` and adopt `--wave_scheme=WAVE_GRAMFE`. Must set `--fftw` for `--gramfe_scheme=GRAMFE_FFT` and `--gpu=false` | <a name="--gramfe_scheme"></a> `GRAMFE_SCHEME` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:65]: - "`GRAMFE_MATMUL`: matrix-multiplication approach, which is faster for `PATCH_SIZE`=8; `GRAMFE_FFT`: FFT approach, which is faster for larger patch sizes"
- "Must set `--fftw` when using both `--gramfe_scheme=GRAMFE_FFT` and `--gpu=false` | <a name="--gramfe_scheme"></a> `GRAMFE_SCHEME`"

File: doc/wiki/Installation:-Option-List.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+| `--hybrid_scheme`    | `HYBRID_UPWIND`, `HYBRID_FROMM`, `HYBRID_MUSCL` | `HYBRID_MUSCL`  | Fluid schemes. `HYBRID_UPWIND`: first-order, diffusive; `HYBRID_FORMM`: second-order, no limiter, unstable for fluid-only simulations; `HYBRID_MUSCL`: second-order, with limiter, useful for zoom-in and fluid-only simulations | Must set `--model=ELBDM` and adopt `--elbdm_scheme=ELBDM_HYBRID` | <a name="--hybrid_scheme"></a> `HYBRID_SCHEME` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:66]: "Fluid schemes supported by the fluid-wave hybrid algorithm"

File: doc/wiki/Installation:-Option-List.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+| `--gravity`  | `true`, `false`  | `false` | Enable [[gravity \| Gravity]] | Must enable `--fftw`; may need to set `FFTW2/3_PATH` in [[configuration file \| Installation:-Machine-Configuration-File#1-Library-paths]] | <a name="--gravity"></a> `GRAVITY` |
+| `--particle` | `true`, `false`  | `false` | Enable [[particles \| Particles]] | Must enable `--gravity` or `--tracer` | <a name="--particle"></a> `PARTICLE` |
+| `--grackle`  | `true`, `false`  | `false` | Enable [[GRACKLE \| Chemistry and Radiation]] | May need to set `GRACKLE_PATH` in [[configuration file \| Installation:-Machine-Configuration-File#1-Library-paths]]; only support `--eos=GAMMA/COSMIC_RAY`; does not support `--comoving` | <a name="--grackle"></a> `SUPPORT_GRACKLE` |
+| `--passive`  | &#8805; 0        | `0`     | Number of user-defined passive scalars | See [[here \| Adding-New-Simulations#v-add-problem-specific-grid-fields-and-particle-attributes]] for details; not supported for `--flu_scheme=RTVD` | <a name="--passive"></a> `NCOMP_PASSIVE_USER` |
+
+## Hydro Options
+-- see [[Hydro]] for the related runtime parameters and other settings
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:42]: Add "; must enable `--model=HYDRO`" at the end of this sentence. Modify other similar descriptions below each section too.

File: doc/wiki/Installation:-Option-List.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+| `--srhd`       | `true`, `false`                                            | `false` | Special relativistic hydrodynamics | Must adopt `--eos=TAUBMATHEWS` | <a name="--srhd"></a> `SRHD` |
+| `--cosmic_ray` | `true`, `false`                                            | `false` | Cosmic rays | Must adopt `--eos=COSMIC_RAY` | <a name="--cosmic_ray"></a> `COSMIC_RAY` |
+| `--eos`        | `GAMMA`, `ISOTHERMAL`, `COSMIC_RAY`, `TAUBMATHEWS`, `USER` | Depend  | [[Equation of state \| equation-of-state]] | The following options only support `GAMMA`: `--flu_scheme=RTVD/CTU`, `--flux=EXACT/ROE`, `--comoving`, `--dual`; see also [[--barotropic \| Installation:-Option-List#--barotropic]] | <a name="--eos"></a> `EOS` |
+| `--barotropic` | `true`, `false`                                            | Depend  | Is [[--eos \| Installation:-Option-List#--eos]] barotropic? | Must be disabled for `--eos=GAMMA` and enabled for `--eos=ISOTHERMAL` | <a name="--barotropic"></a> `BAROTROPIC_EOS` |
+
+## ELBDM Options
+-- see [[ELBDM]] for the related runtime parameters and other settings
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:57]: Add "; must enable --model=ELBDM" at the end of this sentence. Then you can remove "Must set `--model=ELBDM`" from the `Restriction` column.

File: doc/wiki/Installation:-Option-List.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+|:---:|:---:|:---:|---|---|---|
+| `--openmp`      | `true`, `false` | `true` | Enable OpenMP (see [[MPI and OpenMP]]) | Must set the compilation flag `OPENMPFLAG` in [[configuration file \| Installation:-Machine-Configuration-File#3-Compilation-flags]] | <a name="--openmp"></a> `OPENMP` |
+| `--mpi`         | `true`, `false` | `false` | `true`: Enable load balancing using a space-filling curve (see [[MPI and OpenMP]]); `false`: Run GAMER in a serial mode, but OpenMP is still supported | May need to set `MPI_PATH` in [[configuration file \| Installation:-Machine-Configuration-File#1-Library-paths]] | <a name="--mpi"></a> `LOAD_BALANCE=HILBERT`, <a name="SERIAL"></a> `SERIAL` |
+| `--overlap_mpi` | `true`, `false` | `false` | Overlap MPI communication with computation. | Not supported yet!!! Must enable `--mpi`. | <a name="--overlap_mpi"></a> `OVERLAP_MPI` |
+| `--gpu`         | `true`, `false` | `false` | Enable GPU acceleration | Must specify `GPU_COMPUTE_CAPABILITY` in configuration file as well; may need to set `CUDA_PATH` in [[configuration file \| Installation:-Machine-Configuration-File#1-Library-paths]] | <a name="--gpu"></a> `GPU` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:115]: Change to "Must specify GPU_COMPUTE_CAPABILITY and may need to set CUDA_PATH in configuration file"

File: doc/wiki/Installation:-Option-List.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+| `--fftw`                    | `OFF`, `FFTW2`, `FFTW3`    | `OFF`         | Enable FFTW | May need to set `FFTW2/3_PATH` in [[configuration file \| Installation:-Machine-Configuration-File#1-Library-paths]] | <a name="--fftw"></a> `SUPPORT_FFTW` |
+| `--spectral_interpolation`  | `true`, `false`            | `false`       | Enable spectral interpolation | Must enable `--gsl` and set `--fftw` | <a name="--spectral_interpolation"></a> `SUPPORT_SPECTRAL_INT` |
+| `--rng`                     | `RNG_GNU_EXT`, `RNG_CPP11` | `RNG_GNU_EXT` | Random number generators. `RNG_GNU_EXT`: GNU extension `drand48_r`; `RNG_CPP11`: c++11 `&lt;random&gt;` | Use `RNG_GNU_EXT` for compilers supporting GNU extensions (<b>they may not be supported on macOS</b>); use `RNG_CPP11` for compilers supporting c++11 (<b>one may need to add `-std=c++11` to `CXXFLAG`</b> &#8594; see [[compilation flags \| Installation:-Machine-Configuration-File#3-Compilation-flags]]) | <a name="--rng"></a> `RANDOM_NUMBER` |
```
Comment (hyschive) [inline doc/wiki/Installation:-Option-List.md:136]: - It displays as `&lt;random&gt;` instead of `<random>`

**Review by hyschive — CHANGES_REQUESTED** [review]
@ChunYen-Chen Looks great! I only have some minor comments. A few links do not work correctly.

**Review by hyschive — APPROVED** [review]
Everything looks good!

**Pull request merged**